src/elasticsearch_cleanup/utils.py

```diff
@@ -0,0 +1,73 @@
+from typing import Any, Iterable, Tuple
+
+
+def str_to_bool(string: str) -> bool:
+    """Converts a string to its boolean representation.
+
+    Args:
+        string: The string to be converted.
+
+    Returns:
+        The boolean representation of the string.
+
+    Raises:
+        ValueError: If the string does not have a valid boolean representation.
+    """
+    true_values = ["true", "yes", "y", "1", "t"]
+    false_values = ["false", "no", "n", "0", "f"]
+
+    normalized = string.strip().lower()
+
+    if normalized in true_values:
+        return True
+    elif normalized in false_values:
+        return False
+    else:
+        raise ValueError(f"Cannot convert {string} to a bool")
+
+
+def get_value_from_dict(dot_notation_string: str, data: dict[str, Any]) -> Any:
+    """Retrieves a value from a dictionary based on dot notation string.
+
+    Args:
+        dot_notation_string: The string in dot notation representing keys in the dictionary.
+        data: The dictionary to extract data from.
+
+    Returns:
+        The value extracted from the dictionary based on the dot notation string.
+
+    Raises:
+        ValueError: If any key in the dot notation string is not found in the dictionary.
+    """
+    keys = dot_notation_string.split(".")
+
+    for key in keys:
+        if key in data.keys():
+            data = data[key]
+        else:
+            raise ValueError(f"Key {key} not found in the dictionary")
+
+    return data
+
+
+def parse_index_and_fields_pairs(pairs: str) -> Iterable[Tuple[str, list[str]]]:
```

Comment:
```suggestion
def parse_index_and_fields_pairs(pairs: str) -> Iterable[tuple[str, list[str]]]:
```
indtead of a `tuple[str, list[str]]` if could be worth to have a dataclass instead to make things clearer, e.g.:
```
@dataclass
class IndexFieldSelector:
  index: str
  field_selection: list[str]
```